plugins/outputs/iotdb/iotdb.go

```diff
@@ -229,6 +231,60 @@ func (s *IoTDB) convertMetricsToRecordsWithTags(metrics []telegraf.Metric) (*rec
 	return rwt, nil
 }
 
+// takes in an array of compiled regex patterns
+// and validates them on the given tag.
+func matchToEncloseInBacktick(regexStrings []*regexp.Regexp, tag string) string {
+	// loops through all the regex patterns and if one
+	// pattern matches returns the tag between `
+	for _, regex := range regexStrings {
+		if regex.MatchString(tag) {
+			return "`" + tag + "`"
+		}
+	}
+
+	return tag
+}
+
+// checks is the tag contains any IoTDB invalid character
+func (s *IoTDB) validateTag(tag string) (string, error) {
+	// matches any word that has a non valid backtick
+	// `word`  							 <- dosen't match
+	// ``word , `wo`rd` , `word , word`  <- match
+	forbiddenBacktick, _ := regexp.Compile("^[^\x60].*?[\x60]+.*?[^\x60]$|^[\x60].*[\x60]+.*[\x60]$|^[\x60]+.*[^\x60]$|^[^\x60].*[\x60]+$")
+	allowedBacktick, _ := regexp.Compile("^[\x60].*[\x60]$")
+
+	// IoTDB uses "root" as a keyword and can be called only at the start of the path
+	if tag == "root" {
+		return "", fmt.Errorf("can't append keyword \"root\" to IoTDB base path")
```

Comment:
Should this error message actually read that you "cannot use \"root\" as tag?

```suggestion
		return "", errors.New("cannot append keyword \"root\" to IoTDB base path")
```